### ISSUE TITLE
`find` incorrectly converts records to strings

### ISSUE THREAD

**Issue by KAAtheWiseGit** [nushell/nushell#13898]
### Describe the bug

When `find` is used on a table with nested records, it doesn't add a space between key-value pairs.

### How to reproduce

```Nushell
~> [[column]; [{key1: val1, key2: val2}]] | find val
───┬────────────────────────
 # │         column         
───┼────────────────────────
 0 │ {key1: val1key2: val2} 
───┴────────────────────────
```

### Expected behavior

A space in the result `{key1: val1 key2: val2}`.  Alternatively, `find` can avoid flattening the table.

### Configuration

| key                | value                                                        |
| ------------------ | ------------------------------------------------------------ |
| version            | 0.98.0                                                       |
| major              | 0                                                            |
| minor              | 98                                                           |
| patch              | 0                                                            |
| branch             |                                                              |
| commit_hash        |                                                              |
| build_os           | linux-x86_64                                                 |
| build_target       | x86_64-alpine-linux-musl                                     |
| rust_version       | rustc 1.81.0 (eeb90cda1 2024-09-04) (Alpine Linux 1.81.0-r0) |
| cargo_version      | cargo 1.81.0 (2dbb1af80 2024-08-20)                          |
| build_time         | 2024-09-22 02:53:40 +00:00                                   |
| build_rust_channel | release                                                      |
| allocator          | mimalloc                                                     |
| features           | default, sqlite, trash                                       |
| installed_plugins  |

**Comment by sholderbach** [comment]
I am inclined to say that this whole string coercion in `find` is a bit short sighted if `find` should be used in a programmatic context. Either we drop the nested search entirely or handle that highlighting in `table` in a smart way, so the raw data can still be passed on to other commands.

**Comment by KAAtheWiseGit** [comment]
I agree here.  Although, if `find` is to be used as plumbing, it needs a flag to disable highlighting with escape sequences

**Comment by sholderbach** [comment]
I don't want a flag. I want `find` to use `PipelineMetadata` to communicate its highlighting intent and `table` to do the actual highlighting afterwards. (alternative would be that we provide a way to commands to know if they are part of a pipeline, in the oldscool `atty` way.)

**Comment by KAAtheWiseGit** [comment]
I didn't think about moving highlighting to `table`, but it does seem elegant.  `PipelineMetadata` is quite limited right now, though
